### PR TITLE
[Mellanox] Facilitate automatic integration of sdk kernel patches

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Facilitate Automatic integration of sdk kernel patches into SONiC. 

**Inputs to the Script:**
1) `MLNX_SDK_VERSION` Eg: `4.5.4206`
2) `MLNX_SDK_ISSU_VERSION` Eg: `101` 
 **Note: If nothing is provided the one already present in the sdk.mk file is used**
3) `MLNX_SDK_SOURCE_BASE_URL:` 
 **Note: If nothing is provided the upstream sdk drivers url is used**
4) `CREATE_BRANCH: (y|n)` Creates a branch instead of a commit (optional, default: n) 
5) `BRANCH_SONIC`:  Only relevant when CREATE_BRANCH is y. `Default: master`. 

Note: These should be provided through `SONIC_OVERRIDE_BUILD_VARS ` parameter

**Output:**
1) Script creates a commit in sonic-linux-kernel with any updates to sdk-kernel patches in sonic in accordance with the version provided by  `MLNX_SDK_VERSION`

**Note: Script Doesn't commit anything to linux-kernel when there aren't any changes required..**  

#### How I did it

1) Added a new make target which can be invoked by calling `make integrate-mlnx-sdk`

```
user@server:/sonic-buildimage/src/sonic-linux-kernel$ git rev-parse --abbrev-ref HEAD
master_6f38dca_integrate_4.5.4206

user@server:/sonic-buildimage/src/sonic-linux-kernel$ git log --oneline -n 1
d64d1e7 (HEAD -> master_6f38dca_integrate_4.5.4206) Intgerate MLNX SDK 4.5.4206 Kernel Patches
```

Changes made will be summarized under `sonic-buildimage/integrate-mlnx-sdk_user.out` file. Debugging and troubleshooting output is written to `sonic-buildimage/integrate-mlnx-sdk.log` files

[log_files.zip](https://github.com/sonic-net/sonic-buildimage/files/11226441/log_files.zip)

#### Limitations:
1) Assumes that the sdk kernel patches are always upstreamed

#### How to verify it

Build the Kernel and test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

**Review by dgsudharsan — DISMISSED** [review]

**Review by dgsudharsan — APPROVED** [review]

File: platform/mellanox/sdk.mk [hunk 1]
```diff
@@ -17,6 +17,7 @@
 MLNX_SDK_VERSION = 4.5.4206
 MLNX_SDK_ISSU_VERSION = 101
 
+MLNX_SDK_DRIVERS_GITHUB_URL = https://github.com/Mellanox/Spectrum-SDK-Drivers
```
Comment (qiluo-msft) [inline platform/mellanox/sdk.mk:20]: >MLNX_SDK_DRIVERS_GITHUB_URL [](http://example.com/codeflow?start=0&length=27)

Could you reuse this in MLNX_SX_KERNEL_GITHUB_URL_BASE?

File: platform/mellanox/integration-scripts/helper.py [hunk 2]
```diff
@@ -169,3 +171,22 @@ def perform(self):
...
+        return ""
+    content = "\n"
+    content = content + " ## Patch List\n"
+    for key, value in changes.items():
+        content = content + f"* {key} : {value}\n"
+    return content
+
+def parse_id(id_):
+    if id_ and id_ != "N/A":
+        id_ = "https://github.com/gregkh/linux/commit/" + id_
```
Comment (qiluo-msft) [inline platform/mellanox/integration-scripts/helper.py:187]: >gregkh [](http://example.com/codeflow?start=34&length=6)

This is a personal repo. Why?

Comment (vivekrnv) [reply platform/mellanox/integration-scripts/helper.py:187]: No, this is the stable tree mirror

File: platform/mellanox/sdk.mk (hunk 1 above)
Comment (vivekrnv) [reply platform/mellanox/sdk.mk:20]: Actually, i can use MLNX_SX_KERNEL_GITHUB_URL_BASE directly in my code. no need for a different variable. Thanks for noticing

Comment (vivekrnv) [reply platform/mellanox/sdk.mk:20]: Sry, you original suggestion is preferable. I've updated it

**Review by dgsudharsan — DISMISSED** [review]

File: platform/mellanox/integration-scripts/helper.py (hunk 2 above)
Comment (vivekrnv) [reply platform/mellanox/integration-scripts/helper.py]: Updated

**Review by dgsudharsan — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by vivekrnv** [comment]
@yxieca , Why do we need this feature for 202205?

**Comment by yxieca** [comment]
> @yxieca , Why do we need this feature for 202205?

My bad, I mistakenly thought the conflict was for 202205.

**Pull request merged**